File: src/main/java/com/rover12421/csdn/java2kt/java/_7_NegativeLiteral.java
package com.rover12421.csdn.java2kt.java;

/**
 * Created by rover12421 on 8/11/16.
 */
public class _7_NegativeLiteral {
    public static int covertKt(int x){
        String compX = Integer.toBinaryString(x);
        if (x < 0) {
            compX = "-" + Integer.toHexString(~x+1);
        }

        System.out.println("x(" + Integer.toHexString(x) + ") covertKt is " + compX);
        return Integer.parseInt(compX, 16);
    }

    public static void main(String[] args) {
        int a =  0x80000001;
        int b = -0x7FFFFFFF;
        System.out.println("a==b ?= " + (a == b));

        long al =  0x8000000000000001L;
        long bl = -0x7fffffffffffffffL;
        System.out.println("al==bl ?= " + (al == bl));

        System.out.println("a = covertKt(a) ?= " + (a == covertKt(a)));
    }
}

File: src/main/kotlin/com/rover12421/csdn/java2kt/kt/_7_NegativeLiteral.kt
package com.rover12421.csdn.java2kt.kt

import java.math.BigInteger

/**
 * Created by rover12421 on 8/11/16.
 */
object _7_NegativeLiteral {
    @JvmStatic fun main(args: Array<String>) {
        val a:Int = 0x80000001.toInt()
        val b:Int = -0x7FFFFFFF
        println("a==b ?= " + (a == b))

        val al = BigInteger("8000000000000001", 16).toLong()
        val bl = -0x7fffffffffffffffL
        println("al==bl ?= " + (al == bl))

    }
}
